src/handlebars-utils.js

```diff
@@ -177,6 +182,26 @@ HandlebarsUtils.isValidExpression = function(input, i, type) {
     return re;
 };
 
+// @function HandlebarsUtils.parseEscapeExpression
+HandlebarsUtils.parseEscapeExpression = function(str) {
+  var j=0, inSquareBracket = false,
+      fstr = '', re = [];
+
+  str = str.trim(); // the new regexp will capture the tailing space.
```

Comment:
String.trim() is unfriendly to IE 8 or below
https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/String/trim